### ISSUE TITLE
[Bug]: Button group widget's pop-over/drop-down enlarged along the width

### ISSUE THREAD

**Issue by keyurparalkar** [appsmithorg/appsmith#11564]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When we turn the button inside a button group widget into a menu type, then the pop-over/dropdown of that menu button has an enlarged width. 

Below is the image for reference:
![Screenshot 2022-03-02 at 4 43 42 PM](https://user-images.githubusercontent.com/14138515/156355578-27378de9-665d-4802-9f66-2acfec4a6bd1.png)

### Steps To Reproduce

1. Drag and drop the button group widget on to the canvas.
2. Change one of the buttons to menu type.
3. Add some menu items to it.
4. Then click on the button.
5. Observe.

### Environment

Release

### Version

Cloud

**Comment by dilippitchika** [comment]
On a side note - Why did we design the menu button to occupy width of the group and not limit it to the width of button?

**Comment by keyurparalkar** [comment]
@dilippitchika not sure about this. But I think it might be related to this issue: https://github.com/appsmithorg/appsmith/issues/6609

> On a side note - Why did we design the menu button to occupy width of the group and not limit it to the width of button?

**Comment by Tooluloope** [comment]
@dilippitchika is there a design for this, so we can correct it?

**Comment by Nikhil-Nandagopal** [comment]
@Tooluloope it should just be as large as the button, not the button group. That should be a simple solve right?

**Comment by dilippitchika** [comment]
@Tooluloope what Nikhil mentioned should be the fix. If you also agree, please move this to product backlog and assign this to the team.

**Comment by wmdev0808** [comment]
@somangshu,
Based on @Nikhil-Nandagopal's comments, I've just simply put the estimation to 0.5
However, while I was working on this issue, I've realized it is not such a simple fix.
Here are the reasons:
- We need to deal with multiple menu buttons of which width should dynamically be calculated with all possible cases(property changes, resize, etc.), not with single element.
- For this, we need to introduce multiple `refs` to keep track of these dynamic menu button elements.
To sum up, its complexity has been increased more than originally expected one.
I've already spent more than 0.5 and I am sure I will be able to complete it with additional 1.
So the final estimation for this issue should be 1.5, I think.
I need your approval.
What do you think of my request?

**Comment by dilippitchika** [comment]
@wmdev0808 as discussed please use the min-width from the menu button widget here to prevent small menus when button size is small

**Comment by btsgh** [comment]
As per earlier comment by @dilippitchika - "as discussed please use the min-width from the menu button widget here to prevent small menus when button size is small"

This issue looks to be resolved.

<img width="416" alt="Bug11564_1" src="https://user-images.githubusercontent.com/101863839/160636179-37f9190b-9981-44d7-b984-13ceb194e8db.png">
<img width="267" alt="Bug11564_2" src="https://user-images.githubusercontent.com/101863839/160636187-59f9a89e-35b5-4718-90ca-21dab2d133b1.png">
